src/main.rs

```diff
@@ -8,26 +9,36 @@ mod config;
 mod rules;
 
 fn main() {
-    for arg in env::args().skip(1) {
-        check_file(&arg);
+    let mut args: Vec<String> = env::args().skip(1).collect();
+    if args.is_empty() {
+        args = vec![String::from(".")]
+    }
+    for arg in args.iter() {
+        for entry in walkdir::WalkDir::new(arg) {
+            match entry {
+                Err(err) => eprintln!("{:}", err),
+                Ok(entry) if entry.file_type().is_file() => check_file(entry.path()),
+                Ok(_) => {}
+            }
+        }
     }
 }
 
-fn check_file(arg: &str) {
-    let src = fs::read_to_string(arg).unwrap();
-    let nsrc = NamedSource::new(arg, src.clone());
+fn check_file(path: &Path) {
+    let Some(lang) = detect_language(path) else {
+        return;
+    };
+    let filename = path.display().to_string();
+    let src = fs::read_to_string(path).unwrap();
+    let nsrc = NamedSource::new(&filename, src.clone());
```

Comment:
IMHO, we should not rely on semi-obscure abbreviations. I much prefer we have long variable names over hard-to-read code. I suggest we expand the variable name here to `named_source`.